src/AppNavigator.tsx

```diff
@@ -3,119 +3,123 @@ import {NavigationContainer, useNavigationContainerRef} from '@react-navigation/
+    if (isReady && (authAction !== 'os' || !hasAnyWallet)) {
+      RNBootSplash.hide({
+        fade: true,
+      })
+    }
+  }, [authAction, hasAnyWallet, isReady])
+
+  React.useEffect(() => {
+    if (authAction === 'os' && !isEmptyString(secretKey) && isLoggedOut && hasAnyWallet) {
+      loadSecret({
+        key: secretKey,
+        authenticationPrompt: {
+          cancel: strings.cancel,
+          title: strings.authorize,
+        },
+      })
+    }
+  }, [authAction, hasAnyWallet, isLoggedOut, loadSecret, secretKey, strings.authorize, strings.cancel])
+
+  if (authAction == null) return null
+  if (isEmptyString(secretKey)) return null
+
+  return (
+    <NavigationContainer onReady={() => setIsReady(true)} ref={navRef}>
+      {IS_STORYBOOK ? <StoryBook /> : <NavigatorSwitch authAction={authAction} />}
+    </NavigationContainer>
+  )
 }
 
 export default AppNavigator
 
 const Stack = createStackNavigator<AppRoutes>()
-const NavigatorSwitch = () => {
+const NavigatorSwitch = ({authAction}: {authAction: 'create-link-pin' | 'check-pin' | 'os'}) => {
   const strings = useStrings()
   const isMaintenance = useSelector(isMaintenanceSelector)
-  const isSystemAuthEnabled = useSelector(isSystemAuthEnabledSelector)
   const hasAnyWallet = useSelector(hasAnyWalletSelector)
-  const isAppSetupComplete = useSelector(isAppSetupCompleteSelector)
-  const canEnableBiometrics = useSelector(canEnableBiometricSelector)
+  const isTosAccepted = useSelector(isTosAcceptedSelector)
   const installationId = useSelector(installationIdSelector)
-  const queryClient = useQueryClient()
-  const dispatch = useDispatch()
-  const {isLoggedIn, isLoggedOut, login, logout} = useAuth()
+  const {isLoggedIn, isLoggedOut} = useAuth()
 
   if (isEmptyString(installationId)) throw new Error('invalid state')
 
-  useEffect(() => {
-    const appStateSubscription = AppState.addEventListener('change', async () => {
-      await dispatch(checkBiometricStatus(logout))
-      queryClient.invalidateQueries(['walletMetas'])
-    })
-    return () => appStateSubscription?.remove()
-  }, [dispatch, logout, queryClient])
-
-  useEffect(() => {
-    if (hasAnyWallet && isLoggedOut && isSystemAuthEnabled && !canEnableBiometrics && !isMaintenance) {
-      Alert.alert(strings.biometricsChangeTitle, strings.biometricsChangeMessage)
-    }
-  }, [hasAnyWallet, isLoggedOut, isSystemAuthEnabled, canEnableBiometrics, isMaintenance, strings])
-
   return (
     <Stack.Navigator screenOptions={{headerShown: false}}>
       {/* Initial Route */}
       <Stack.Group>
         {isMaintenance && <Stack.Screen name="maintenance" component={MaintenanceScreen} />}
-        {!isAppSetupComplete && !hasAnyWallet && <Stack.Screen name="first-run" component={FirstRunNavigator} />}
+        {!isTosAccepted && <Stack.Screen name="first-run" component={FirstRunNavigator} />}
       </Stack.Group>
 
       {/* Not Authenticated */}
       {isLoggedOut && hasAnyWallet && (
```

Comment:
why does having wallets affect the logged out ui?

Reply:
it was designed for UX, so the user is not prompted for authentication when there is no wallet, I'd like to change it as well.